pulumi/python/automation/main.py

```diff
@@ -85,28 +96,45 @@ def usage():
     print(usage_text, file=sys.stdout)
 
 
-def provider_instance(provider_name: str) -> Provider:
-    """Dynamically instantiates an infrastructure provider
-    :param provider_name: name of infrastructure provider
-    :return: instance of infrastructure provider
-    """
-    module = importlib.import_module(name=f'providers.{provider_name}')
-    return module.INSTANCE
+def write_env(env_config, stack_name):
```

Comment:
Optional, but if you only need `filename` consider just passing in `filename`.  If this module exclusively dealt with the `env_config` object then I could see having functions just take in that object and use fields on it but since this is a general module it would be simpler and less error prone not to force this function to have knowledge of the keys on `env_config`